landscape_setup/concourse.py

```diff
@@ -181,6 +213,67 @@ def create_instance_specific_helm_values(
         raise NotImplementedError(
             "Concourse version {v} not supported".format(v=concourse_version)
         )
+    # Add proxy sidecars to instance specific values.
+    # NOTE: Only works for helm chart version 4.2.2 or greater
+    if concourse_cfg.proxy():
+        instance_specific_values = add_proxy_values(
+            concourse_cfg=concourse_cfg,
+            config_factory=config_factory,
+            instance_specific_values=instance_specific_values,
+        )
+
+    return instance_specific_values
+
+
+def add_proxy_values(
+    concourse_cfg: ConcourseConfig,
+    config_factory: ConfigFactory,
+    instance_specific_values: dict,
+):
+    # add the sidecar-configuration for the mitm-proxy
+    proxy_cfg = config_factory.proxy(concourse_cfg.proxy())
+    mitm_cfg = proxy_cfg.mitm_proxy()
+    iptables_setup_cfg = proxy_cfg.setup_iptables()
+    sidecar_containers = [{
+        'name': 'setup-iptables-sidecar',
+        'image': f'{iptables_setup_cfg.image_name()}:{iptables_setup_cfg.image_tag()}',
+        'env': [{
+            'name': 'PROXY_PORT',
+            'value': f'{mitm_cfg.config()["listen_port"]}',
+        }],
+        'securityContext': {
+            'privileged': True,
+        },
+    },{
+        'name': 'mitm-proxy',
+        'image': f'{mitm_cfg.image_name()}:{mitm_cfg.image_tag()}',
+        'env': [{
+                'name': 'CONFIG_DIR',
+                'value': MITM_CONFIG_DIR,
+        }],
+        'ports': [{
+            'containerPort': mitm_cfg.config()["listen_port"],
+            'hostPort': mitm_cfg.config()["listen_port"],
+            'protocol': 'TCP',
+        }],
+        'volumeMounts': [{
+            'name': 'mitm-config',
+            'mountPath': MITM_CONFIG_DIR,
+        }],
+    }]
+    additional_volumes = [{
+        'name':'mitm-config',
+        'configMap': {'name': MITM_CONFIG_CONFIGMAP_NAME},
+    }]
+    # add new values to dict without replacing existing ones
+    vals = instance_specific_values.get('worker', {})
+    vals.update(
```

Comment:
why not just assign the attributes (vals['sidecarContainers'] = sidecar_containers..)